src/GraphQL/Types/ComplexGraphType.cs

```diff
@@ -0,0 +1,82 @@
+using GraphQL.Builders;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace GraphQL.Types
+{
+    public abstract class ComplexGraphType : GraphType {
+        private readonly List<FieldType> _fields = new List<FieldType>();
+
+        public List<FieldType> Fields
```

Comment:
This exposes the underlying `List` to the outside world.  I have been trying to keep that encapsulated within the class so that the user is forced to use one of the api methods and doesn't have direct access to the list.  Doing so makes it so some of the rules (such as unique field names) get applied.

`IEnumerable<FieldType>` vs. `List<FieldType>`